### PR TITLE
Trim leading and trailing spaces in name

### PR DESCRIPTION
**Pull request by KateMengjunXia**
https://nanopay.atlassian.net/browse/CPF-725

### CODE REVIEW

File: src/foam/core/Property.js [hunk 1]
```diff
@@ -522,6 +523,7 @@ foam.CLASS({
           if ( assertValue ) assertValue.call(this, newValue, prop);
 
           if ( preSet ) newValue = preSet.call(this, oldValue, newValue, prop);
+          if ( trim ) newValue = newValue.trim();
```
Comment (kgrgreer) [inline src/foam/core/Property.js]: I think this should go before adapt and assertValue. Do you agree?

Comment (KateMengjunXia) [reply src/foam/core/Property.js]: Yes

**Comment by jlhughes** [comment]
@kgrgreer Sounds like the original intent was to default to true.  If we break something with default: true, then it was already broken.

**Comment by KateMengjunXia** [comment]
> @kgrgreer Sounds like the original intent was to default to true. If we break something with default: true, then it was already broken.

Great catch. It seems like when default to true, there are cases where oldValue is a string and newvalue is of some other type, and trim doesn't work.

**Comment by jlhughes** [comment]
OK, so with the additional tests for type 'string', can we switch this to default: true?

File: src/foam/core/Property.js (hunk 1 above)
Comment (mcarcaso) [reply src/foam/core/Property.js]: Sorry for the retro review here, but why have trim baked into foam.core.Property? Isn't this just a String concept? We generate an adapt on String that should do this work.

Comment (mcarcaso) [reply src/foam/core/Property.js]: See: https://github.com/foam-framework/foam2/pull/2106/files

**Pull request merged**